### ISSUE TITLE
SVDquartets (PAUP version) renames the taxa

### ISSUE THREAD

**Issue by j-chou** [j-chou/svdquartets#1]
All the scripts so far work on the cc. Using get_topology_trees.sh we can run the SVDquartets tp QMC/wQMC pipeline on files of DNA alignments in Nexus format. For the data we have, there are 11 taxa for each file labeled 0-10. However, SVDquartets in PAUP renames the taxa from 1-11 and I'm not sure whether the mapping is n -> n+1. This causes QMC and wQMC (max-cut-tree) to return Newick trees with leaves labeled 1-11, so at this point we cannot make comparisons between the true species tree which has leaves labeled 0-10 with the estimated trees from the SVDquartets to QMC/wQMC pipeline which has leaves labeled from 1-11.

**Comment by hyaduvanshi** [comment]
We should be able to find that out by going into what paup does. I will return tomorrow so I can try checking that once I am back. Jed, did you manage to find some good candidates for the mapping from svdquartet to wqmc? 
Thanks
Shashank

**Comment by j-chou** [comment]
Yes, sort of. I'm mimicking what they do in the weighted quartets phylogenetics paper to get a measure of how reliable each quartet is. In the paup_to_qmc_wqmc file I define a "reliability score". 
Also, I think there's a bug in PAUP for the file conversions. I'm unable to convert any of the phylip files into nexus through PAUP, so I had to find a python library called bioscripts to do it. The problem is that it's extremely slow and we have more than a million files to convert. Also, cc I don't think allows us to install these libraries. I'm in the process of converting 10000 files on my machine and will send them to you guys tonight.

**Comment by redavids** [comment]
I have been thinking about this and I've decided we must prioritize getting a script (python or whatever) working that takes as an argument a dictionary such as {'bug': '1', 'cat': '2'} as an argument to rewrite taxon names a file at a time.  It is not realistic that we'll be working with files where we don't know a priori what the taxon set is, so this is practical.  Modifying taxonfixer.py at 

https://github.com/redavids/phylogenetics-tools

should do this.  Currently taxonfixer.py takes a file and returns a dict showing the taxon name mapping as well as taxon names changed to integers: we want to provide the dict as input so we control the conversion.  For example this will allow us to bypass the 0-10 vs 1-11 issue because we can do the mapping n->n+1 first.  I started writing some tests first for this but it should be easy to modify the idea in taxonfixer.py.

**Comment by redavids** [comment]
have been thinking about this and I've decided we must prioritize getting
a script (python or whatever) working that takes as an argument a
dictionary such as {'bug': '1', 'cat': '2'} as an argument to rewrite taxon
names a file at a time. It is not realistic that we'll be working with
files where we don't know a priori what the taxon set is, so this is
practical. Modifying taxonfixer.py at

https://github.com/redavids/phylogenetics-tools

should do this. Currently taxonfixer.py takes a file and returns a dict
showing the taxon name mapping as well as taxon names changed to integers:
we want to provide the dict as input so we control the conversion. For
example this will allow us to bypass the 0-10 vs 1-11 issue because we can
do the mapping n->n+1 first. I started writing some tests first for this
but it should be easy to modify the idea in taxonfixer.py.



On Sun, Mar 29, 2015 at 8:20 PM, j-chou <notifications@github.com> wrote:

> Yes, sort of. I'm mimicking what they do in the weighted quartets
> phylogenetics paper to get a measure of how reliable each quartet is. In
> the paup_to_qmc_wqmc file I define a "reliability score".
> Also, I think there's a bug in PAUP for the file conversions. I'm unable
> to convert any of the phylip files into nexus through PAUP, so I had to
> find a python library called bioscripts to do it. The problem is that it's
> extremely slow and we have more than a million files to convert. Also, cc I
> don't think allows us to install these libraries. I'm in the process of
> converting 10000 files on my machine and will send them to you guys tonight.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/j-chou/svdquartets/issues/1#issuecomment-87507415>.
>



-- 
Ruth Davidson
Department of Mathematics, University of Illinois Urbana-Champaign
http://www.math.uiuc.edu/~redavid2/

**Comment by redavids** [comment]
Once we get the taxon name issue resolved and files converted, we'll be
able to compare to true species tree for your data set, correct?

On Tue, Mar 31, 2015 at 8:10 AM, Ruth Davidson <redavids@ncsu.edu> wrote:

>  have been thinking about this and I've decided we must prioritize getting
> a script (python or whatever) working that takes as an argument a
> dictionary such as {'bug': '1', 'cat': '2'} as an argument to rewrite taxon
> names a file at a time. It is not realistic that we'll be working with
> files where we don't know a priori what the taxon set is, so this is
> practical. Modifying taxonfixer.py at
>
> https://github.com/redavids/phylogenetics-tools
>
> should do this. Currently taxonfixer.py takes a file and returns a dict
> showing the taxon name mapping as well as taxon names changed to integers:
> we want to provide the dict as input so we control the conversion. For
> example this will allow us to bypass the 0-10 vs 1-11 issue because we can
> do the mapping n->n+1 first. I started writing some tests first for this
> but it should be easy to modify the idea in taxonfixer.py.
>
>
>
> On Sun, Mar 29, 2015 at 8:20 PM, j-chou <notifications@github.com> wrote:
>
>> Yes, sort of. I'm mimicking what they do in the weighted quartets
>> phylogenetics paper to get a measure of how reliable each quartet is. In
>> the paup_to_qmc_wqmc file I define a "reliability score".
>> Also, I think there's a bug in PAUP for the file conversions. I'm unable
>> to convert any of the phylip files into nexus through PAUP, so I had to
>> find a python library called bioscripts to do it. The problem is that it's
>> extremely slow and we have more than a million files to convert. Also, cc I
>> don't think allows us to install these libraries. I'm in the process of
>> converting 10000 files on my machine and will send them to you guys tonight.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/j-chou/svdquartets/issues/1#issuecomment-87507415>.
>>
>
>
>
> --
> Ruth Davidson
> Department of Mathematics, University of Illinois Urbana-Champaign
> http://www.math.uiuc.edu/~redavid2/
>
>
>


-- 
Ruth Davidson
Department of Mathematics, University of Illinois Urbana-Champaign
http://www.math.uiuc.edu/~redavid2/

**Comment by j-chou** [comment]
But the script taxonfixer.py takes in Newick files right? For this case, we need to convert taxon names for phylip or nexus alignments to integers. So in an alignment we need the first sequence to be mapped to 1, the second sequence name mapped to 2, and so on. I ran a few trials on the non-PAUP SVDQuartets with phylip alignments having only 4 sequences and this is how they do the mapping.

**Comment by j-chou** [comment]
Yes, once we get the taxon name issue resolved we can compare the true species trees to the estimated ones. I just converted one of the phylip alignments by hand and ran the pipeline. Missing branch lengths were as follows:
SVDQuartets+QMC: 0.75
SVDQUartets+wQMC: 0.5
ASTRAL on True Gene Trees: 0
